providers/aws/sns/topics.go

```diff
@@ -0,0 +1,142 @@
+package sns
+
+import (
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/mitchellh/mapstructure"
+	"go.uber.org/zap"
+)
+
+type Topic struct {
+	ID                        uint `gorm:"primarykey"`
+	AccountID                 string
+	Region                    string
+	Owner                     *string
+	TopicArn                  *string
+	Policy                    *string
+	DeliveryPolicy            *string
+	DisplayName               *string
+	SubscriptionsConfirmed    *int
+	SubscriptionsDeleted      *int
+	SubscriptionsPending      *int
+	EffectiveDeliveryPolicy   *string
+	FifoTopic                 *bool
+	ContentBasedDeduplication *bool
+}
+
+func (Topic) TableName() string {
+	return "aws_sns_topics"
+}
+
+func GetTopicAttributes(c *Client, TopicArn *string) (map[string]*string, error) {
+	params := &sns.GetTopicAttributesInput{
+		TopicArn: TopicArn,
+	}
+	output, err := c.svc.GetTopicAttributes(params)
+	if err != nil {
+		return nil, err
+	}
+	return output.Attributes, nil
+}
+
+func getOrZero(attrs map[string]*string, keyName string) string {
+	if v, found := attrs[keyName]; found {
+		return *v
+	}
+	return "0"
+}
+
+func (c *Client) transformTopics(values []*sns.Topic) ([]*Topic, error) {
+	var tValues []*Topic
+	for _, value := range values {
+
+		// All topic attributes are returned as a string; we have to handle type conversion
+		output, err := GetTopicAttributes(c, value.TopicArn)
+		if err != nil {
+			return nil, err
+		}
+
+		subsConfirmed, subsConfirmedErr := strconv.Atoi(getOrZero(output, "SubscriptionsConfirmed"))
```

Comment:
@nick123pig the standard convention in Go is just to use `err` no need to use `prefixErr`.

Reply:
ok thank you (this is my first go project, so feel free to nitpick my style :D)

Reply:
resolved in 968bdc6